### ISSUE TITLE
Pytorch PPO exercise results in no grad_fn error.

### ISSUE THREAD

**Issue by tanmayshankar** [openai/spinningup#243]
Hello, 

The PPO (Pytorch) exercise solutions in exercise1_2 lead to an error. 

Upon running the ppo function ppo(env_fn = lambda...), Pytorch returns an error of `element 0 of tensors does not require grad and does not have a grad_fn`. The error comes from the loss_pi.backward() call in the update function of the ppo.py file. 

The loss_pi tensor doesn't have a gradient function, and digging a little bit, it seems like the MLPGaussianActor() and DiagonalGaussianDistribution classes in the exercise1_2 solutions do not provide tensors with gradients as outputs. 

Replacing the DiagonalGaussianDistribution with torch.distributions.MultiVariateNormal and the MLPGaussianActor implementation with the following code seems to fix the issue:  
`class MLPGaussianActor(nn.Module):

    def __init__(self, obs_dim, act_dim, hidden_sizes, activation):
        super().__init__()
        self.mu_net = mlp([obs_dim] + list(hidden_sizes) + [act_dim], activation)      
        self.std_net = mlp([obs_dim] + list(hidden_sizes) + [act_dim], activation)
        self.softplus_activation = torch.nn.Softplus()

    def forward(self, obs, act=None):
        # Create mean and var. 
        mean = self.mu_net(obs)
        standard_deviation = self.softplus_activation(self.std_net(obs))

        # Distribution. 
        dist = torch.distributions.MultivariateNormal(mean, torch.diag_embed(standard_deviation))

        log_prob = None
        if act is not None:
            log_prob = dist.log_prob(act)

        return dist, log_prob`
